analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/PermissiveCors.cs

```diff
@@ -0,0 +1,148 @@
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.Helpers.Trackers;
+
+namespace SonarAnalyzer.Rules.Hotspots
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class PermissiveCors : TrackerHotspotDiagnosticAnalyzer<SyntaxKind>
+    {
+        private const string DiagnosticId = "S5122";
+        private const string MessageFormat = "Make sure this permissive CORS policy is safe here.";
+        private const string AccessControlAllowOriginHeader = "Access-Control-Allow-Origin";
+        private const string AccessControlAllowOriginPropertyName = "AccessControlAllowOrigin";
+        private const string StarConstant = "*";
+        private const int ParameterCount = 2;
+
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+        public PermissiveCors() : base(AnalyzerConfiguration.Hotspot, DiagnosticId, MessageFormat) { }
+
+        public PermissiveCors(IAnalyzerConfiguration configuration) : base(configuration, DiagnosticId, MessageFormat) { }
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            base.Initialize(context);
+
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                if (IsEnabled(compilationContext.Options))
+                {
+                    context.RegisterSyntaxNodeActionInNonGenerated(VisitAttribute, SyntaxKind.Attribute);
+                }
+            });
+        }
+
+        protected override void Initialize(TrackerInput input)
+        {
+            SetupInvocationTracker(Language.Tracker.Invocation, input);
+            SetupObjectCreationTracker(Language.Tracker.ObjectCreation, input);
+        }
+
+        private static void SetupInvocationTracker(InvocationTracker<SyntaxKind> tracker, TrackerInput input)
+        {
+            tracker.Track(input,
+                          tracker.MatchMethod(new MemberDescriptor(KnownType.System_Collections_Generic_IDictionary_TKey_TValue, "Add")),
+                          tracker.MethodHasParameters(ParameterCount),
+                          c => IsFirstArgumentAccessControlAllowOrigin((InvocationExpressionSyntax)c.Node, c.SemanticModel)
+                               && IsSecondArgumentStarString((InvocationExpressionSyntax)c.Node, c.SemanticModel),
+                          tracker.IsIHeadersDictionary());
+
+            tracker.Track(input,
+                          tracker.MatchMethod(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Cors_Infrastructure_CorsPolicyBuilder, "WithOrigins")),
+                          c => ContainsStar(((InvocationExpressionSyntax)c.Node).ArgumentList.Arguments.Select(a => a.Expression), c.SemanticModel));
+
+            tracker.Track(input, tracker.MatchMethod(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Cors_Infrastructure_CorsPolicyBuilder, "AllowAnyOrigin")));
+
+            tracker.Track(input,
```

Comment:
semantically, it makes sense to move this after line 78, to keep Add/Append methods close